acl/acl.go

```diff
@@ -433,6 +441,33 @@ func (p *PolicyACL) KeyRead(key string) bool {
 	return p.parent.KeyRead(key)
 }
 
+// KeyWritePrefix returns if a prefix is allowed to be read
+func (p *PolicyACL) KeyReadPrefix(prefix string) bool {
+	// Look for a matching rule
+	_, rule, ok := p.keyRules.LongestPrefix(prefix)
+
+	// Look if any of our children have a read policy
```

Comment:
This should find the prefix itself, too, right? Seems like we don't need to fetch that separately.